lib/Serialization/Serialization.cpp

```diff
@@ -4525,7 +4525,7 @@ void swift::serialize(ModuleOrSourceFile DC,
 
   bool hadError = withOutputFile(getContext(DC), options.OutputPath,
                                  [&](raw_ostream &out) {
-    SharedTimer timer("Serialization (swiftmodule)");
+    SharedTimer timer("Serialization, swiftmodule");
```

Comment:
Not sure what the assertion is really trying to get at. It says what it's checking but provides no rationale (aside from the string in question being _in_ quotes already, in JSON; maybe the author was trying to check that the string wasn't going to prematurely end the quote?)

Support/Timer.cpp
~~~
  assert(!yaml::needsQuotes(Name) && "TimerGroup name needs no quotes");
  assert(!yaml::needsQuotes(R.Name) && "Timer name needs no quotes");
~~~

Anyway, as with the rest of this patch, I'm on a deadline to get all this stuff working and wanted to keep it as non-invasive as possible. I talked to Matthias about the bit I _did_ need to change in LLVM and his response was along the lines of "I'm going to rewrite all the stats-emitting stuff soon anyways", so I wouldn't spend time polishing it at this point.